packages/hatchify-core/src/dataTypes/integer/index.spec.ts

```diff
@@ -32,11 +32,80 @@ describe("integer", () => {
 
     it("transforms correctly", () => {
       const {
+        setClientPropertyValue,
+        serializeClientPropertyValue,
+        setClientQueryFilterValue,
+        serializeClientQueryFilterValue,
+        setClientPropertyValueFromResponse,
         serializeORMPropertyValue,
         setORMPropertyValue,
         setORMQueryFilterValue,
       } = type.finalize()
 
+      // setClientPropertyValue
+      expect(setClientPropertyValue(-1)).toBe(-1)
+      expect(setClientPropertyValue(0)).toBe(0)
+      expect(setClientPropertyValue(1)).toBe(1)
+      expect(setClientPropertyValue(null)).toBeNull()
+      expect(() =>
+        setClientPropertyValue("invalid" as unknown as number),
+      ).toThrow(new HatchifyCoerceError("as a number"))
+      expect(() => setClientPropertyValue(-Infinity)).toThrow(
+        new HatchifyCoerceError("different than Infinity"),
+      )
+      expect(() => setClientPropertyValue(Infinity)).toThrow(
+        new HatchifyCoerceError("different than Infinity"),
+      )
+      expect(() => setClientPropertyValue(1.1)).toThrow(
+        new HatchifyCoerceError("as multiples of 1"),
+      )
+
+      // serializeClientPropertyValue
+      expect(serializeClientPropertyValue(-1)).toBe(-1)
+      expect(serializeClientPropertyValue(0)).toBe(0)
+      expect(serializeClientPropertyValue(1)).toBe(1)
+      // This function expects valid data, so it won't throw an error.
+      expect(serializeClientPropertyValue(null)).toBeNull()
+      expect(serializeClientPropertyValue(1.1)).toBe(1.1)
+      expect(serializeClientPropertyValue("invalid" as unknown as number)).toBe(
+        "invalid",
+      )
+
+      // setClientQueryFilterValue
+      expect(setClientQueryFilterValue(-1)).toBe(-1)
+      // todo: more tests!
+      // todo: should this handle filter query strings or objects? ie.
+      // `filter[age]=1&filter[name]=bob`
+      // [{ field: "age", operator: "$eq", value: 1 }, { field: "name", operator: "$eq", value: "bob" }]
```

Comment:
@justinbmeyer is the value that is going to be passed into this function expected to be full filter objects, arrays, or full query strings? Or is the expectation that only the _value_ will be passed into this function?

the confluence mentions

>   // Passed  - Any crazy value the client might use to filter
>  // Returns - A value react-rest can understand / use
>  // Throws  - If the data is bad ❓
>  // Example : ?filter[age]=xyz ... xyz => throw "xyz is not a number";

so for the example, not sure if `?filter[age]=xyz` is being passed in or just `xyz`

Reply:
i think so too, otherwise it gets really weird with long filter strings that contain irrelevant fields, like `filter[age]=1&filter[name]=bob`, and this function doesn't know it should be parsing out the value for `age`

I will write the tests under the assumption it's just value, though I guess maybe `value[]` should be supported? if someone is filtering on `age in [1,2,3]` - if the argument is an array i will run coerce on each instance

@justinbmeyer ^ sanity check would be great to make sure i am not out to lunch with assumptions here 😄